api/src/main/java/io/camunda/zeebe/process/test/api/ObjectMapperConfig.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright © 2021 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.camunda.zeebe.process.test.api;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * Shared custom {@link ObjectMapper} configured by the user
+ */
+public abstract class ObjectMapperConfig {
```

Comment:
❓ ❌  I struggle to understand why we need to config class in the api. AFAIk it's only used by the assertions and to create a client to the engine. We also don't expect users to interact with this config class.
The classes in this api module are the classes that get injected into the user's test classes by the extension. The implementations of those interfaces both do something very different as they directly interact with the engine, which in some cases is running in a docker container, and in others it's embedded and easier to access.

I would've expected to configuration class to set in the `assertions` module. Similar to how we keep track of the `RecordStream` in the `BpmnAssert` class.

This removes the need to have this class duplicated 3 times as well. The extension sets the field in the config class. The assertions can access this when they need it.

Reply:
I thought the api was the only place where put shared infos, but I was wrong. This suggestion make the code more clear and efficient. Thank you 🙏